src/blog/dark-mode-with-style-dictionary/index.mdx

```diff
@@ -0,0 +1,1352 @@
+
+```javascript
+const fs = require('fs-extra');
+const template = require('lodash/template');
+
+function(dictionary, config) {
+  const { androidPath, iosPath, buildPath } = config;
+  
+  dictionary.allProperties
+    .filter(token => {
+      return token.attributes.category === `image`
+    })
+    .forEach(token => {
+      const { name, value } = token;
+      
+      // Read the file from the token's value and turn it into a lodash template
+      const src = template( fs.readFileSync(value) );
+      
+      // Generate the final SVG output by passing in the dictionary
+      // to resolve the references.
+      const svg = src(dictionary.properties);
+      
+      // Make sure the directory exists and write the new SVG file
+      const outputPath = `${buildPath||''}${name}.svg`;
+      fs.ensureFileSync(outputPath);
+      fs.writeFileSync(outputPath, svg);
+      console.log(`✔︎  ${outputPath}`);
+
+      // Take the SVG and convert it into Android Vector Drawable format
+
+      // Take the SVG and convert it to a PNG and iOS imageset
+    });
+}
+```
+
+### Output references
+
+The second important part is the [`outputReferences`](https://amzn.github.io/style-dictionary/#/version_3?id=output-references) option that is coming out in [Style Dictionary 3.0](https://amzn.github.io/style-dictionary/#/version_3). Normally, Style Dictionary resolves all references (aka aliases) so that token values in output files are raw values. For example if you had these 2 tokens:
+
+```json
+{
+  "color": {
+    "red": { "value": "#ff0000" },
+    "danger": { "value": "{color.red.value}" }
+  }
+}
+```
+
+and you output an CSS variables file, it would look like this:
+
+```css
+--color-red: #ff0000;
+--color-danger: #ff0000;
+```
+
+Style Dictionary resolves the references and outputs the resolved value of the tokens. However, this does not make good use of languages like CSS where you can have variables reference other variables. The `outputReferences` option allows certain formats to not use the resolved value for references and instead refer to the variable. The CSS variables file becomes:
+
+```css
+--color-red: #ff0000;
+--color-danger: var(--color-red);
+```
+
+With the `outputReferences` option we can now output only the colors that change in the different modes, and keep the references to them. This will make smaller output files and make better use of the platforms.
+
+Now let's look at how we can implement these dark mode tokens. I often say you can achieve anything in Style Dictionary with the right configuration and token setup. This is why I am showing 2 approaches, both work and have the same end result. The difference is in the configuration and the token authoring experience.
```

Comment:
This is why I am showing 2 approaches, both work and have the same end result. > This is why I am showing two approaches. Both work and have the same result.